src/modules/error/hocs/withPayloadProvider.js

```diff
@@ -0,0 +1,92 @@
+// @flow
```

Comment:
Hell :imp:  of a HOC but yeah probably no easier way right now :D

Reply:
I like it though ^^ Streamlines the other components quite a lot